src/abstractdataframe/abstractdataframe.jl

```diff
@@ -405,29 +413,94 @@ describe(df)
 ```
 
 """
-StatsBase.describe(df::AbstractDataFrame) = describe(stdout, df)
-function StatsBase.describe(io, df::AbstractDataFrame)
-    for (name, col) in eachcol(df)
-        println(io, name)
-        describe(io, col)
-        println(io, )
+StatsBase.describe(df::AbstractDataFrame; kwargs...) = describe(stdout, df; kwargs...)
+function StatsBase.describe(io, df::AbstractDataFrame; stats = [:mean, :min, :median, :max, :nmissing, :eltype])
+    # Check that people don't specify the wrong fields. 
+    allowed_fields = [:mean, :std, :min, :q25, :median, :q75, :max, :nunique, :nmissing, :eltype] 
+    if !issubset(stats, allowed_fields) 
+        disallowed_fields = setdiff(stats, allowed_fields)
+        not_allowed = "Field(s) not allowed: $disallowed_fields. "
+        allowed = "Allowed fields are: $allowed_fields."
+        throw(ArgumentError(not_allowed * allowed)) 
     end
-end
 
-function StatsBase.describe(io::IO, X::AbstractVector{Union{T, Missing}}) where T
-    missingcount = count(ismissing, X)
-    pmissing = 100 * missingcount/length(X)
-    if pmissing != 100 && T <: Real
-        show(io, StatsBase.summarystats(collect(skipmissing(X))))
-    else
-        println(io, "Summary Stats:")
+    # Define 4 functions for getting summary statistics 
+    # use a dict because we dont know which measures the user wants
+    function get_stats(col::AbstractArray{<:Real})
+        sumstats = summarystats(col)
+        Dict(
+            :mean => sumstats.mean,
+            :std => Compat.std(col, mean = sumstats.mean),
+            :min => sumstats.min,
+            :q25 => sumstats.q25,
+            :median => sumstats.median,
+            :q75 => sumstats.q75,
+            :max => sumstats.max,
+            :nmissing => nothing,
+            :nunique => nothing,
+            :eltype => eltype(col)
+        )
+    end
+    
+    function get_stats(col::AbstractArray{<:Union{Real, Missing}})
+        nomissing = collect(skipmissing(col))
+        sumstats = summarystats(nomissing)
+        Dict(
+            :mean => sumstats.mean,
+            :std => Compat.std(nomissing, mean = sumstats.mean),
+            :min => sumstats.min,
+            :q25 => sumstats.q25,
+            :median => sumstats.median,
+            :q75 => sumstats.q75,
+            :max => sumstats.max,
+            :nmissing => count(ismissing, col),
+            :nunique => nothing,
+            :eltype => Missings.T(eltype(col))
+        )
+    end
+
+    function get_stats(col::AbstractArray{>:Missing})
+        Dict(
+            :mean => nothing,
+            :std => nothing,
+            :min => nothing,
+            :q25 => nothing,
+            :median => nothing,
+            :q75 => nothing,
+            :max => nothing,
+            :nmissing => count(ismissing, col),
+            :nunique => length(unique(col)),
+            :eltype => Missings.T(eltype(col))
+        )    
+    end 
+    
+    function get_stats(col)
```

Comment:
Drop this method, it will never be called due to the `::AbstractArray{>:Missing}` one above.

Reply:
Are you sure? I think the "any" method is getting called based on the tests. If it weren't being called, we would see `:nmissing` be `0` for the `non_number` column. But instead we get `nothing`.